### ISSUE TITLE
Launch as Junit test not working for junit 5

### ISSUE THREAD

**Issue by miklossy** [eclipse/xtext-eclipse#1343]
This is a follow up issue of https://github.com/eclipse/xtext-eclipse/issues/653#issuecomment-586701835 reported by @kuniss

**Step to reproduce:**
1. Create a fresh new Xtext MyDsl project with `JUnit 5 as Testing Support` and `Gradle as Preferred Build System`
![image](https://user-images.githubusercontent.com/20393472/74606080-c44b2c00-50cd-11ea-8c2a-81076f2987d7.png)
2. Execute the `GenerateMyDsl.mwe2` workflow
3. Run the generated `MyDslParsingTest.xtend` test case within Eclipse:
4. The following exception is thrown:
```
java.lang.NoClassDefFoundError: org/junit/platform/commons/PreconditionViolationException
	at org.eclipse.jdt.internal.junit5.runner.JUnit5TestLoader.createUnfilteredTest(JUnit5TestLoader.java:75)
	at org.eclipse.jdt.internal.junit5.runner.JUnit5TestLoader.createTest(JUnit5TestLoader.java:66)
	at org.eclipse.jdt.internal.junit5.runner.JUnit5TestLoader.loadTests(JUnit5TestLoader.java:53)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:526)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:770)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:464)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210)
Caused by: java.lang.ClassNotFoundException: org.junit.platform.commons.PreconditionViolationException
	at java.net.URLClassLoader.findClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at sun.misc.Launcher$AppClassLoader.loadClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	... 7 more
```
![screencast](https://user-images.githubusercontent.com/20393472/74606198-b8139e80-50ce-11ea-990d-fb733ee29d36.gif)

**Comment by cdietrich** [comment]
to me this looks like that jdt does no longer work with older junit5 versions

e.g. setting

	testCompile 'org.junit.jupiter:junit-jupiter-api:5.5.0'
	testRuntime 'org.junit.jupiter:junit-jupiter-engine:5.5.0'

seems to work.
maybe we should update the junit5 version in the bom and dont care about oxygen people anymore (or maybe oxygen can handle junit 5.5/5.6 too)

**Comment by kuniss** [comment]
Thanks for the prompt and fast processing, guys. As always! :smile:

**Comment by cdietrich** [comment]
unfortunately oxygen3a with junit 5.5.1 gives

```
java.lang.NoClassDefFoundError: org/junit/platform/launcher/core/LauncherFactory
	at org.eclipse.jdt.internal.junit5.runner.JUnit5TestLoader.<init>(JUnit5TestLoader.java:31)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.Class.newInstance(Class.java:442)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.createRawTestLoader(RemoteTestRunner.java:367)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.createLoader(RemoteTestRunner.java:362)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.defaultInit(RemoteTestRunner.java:306)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.init(RemoteTestRunner.java:221)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:205)
Caused by: java.lang.ClassNotFoundException: org.junit.platform.launcher.core.LauncherFactory
	at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	... 11 more

```

=> which death should we die @szarnekow @kthoms 
cc @tivervac you are one reason to keep oxygen ...

**Comment by cdietrich** [comment]
additionally adding testRuntime 'org.junit.platform:junit-platform-launcher:1.5.1' would solve this too but i have no idea what needs to be added and what not with this crappy junit 5

**Comment by cdietrich** [comment]
=> as there seems to be a workaround for oxygen i simply propose to update

**Comment by miklossy** [comment]
@cdietrich Would it make sense to adjust our integration tests in xtext-reference-projects to also catch such cases?

**Comment by cdietrich** [comment]
Yes.

**Comment by cdietrich** [comment]
=> we will go the route to ditch oxygen here with reference to workaround

**Comment by cdietrich** [comment]
i consider this fixed with 2.21

**Comment by miklossy** [comment]
I have just tested your fix @cdietrich and it is working! Thanks a lot for your assistance!
